src/core/identity.cc

```diff
@@ -50,19 +50,26 @@
 namespace i2p {
 namespace data {
 
+// TODO(unassigned): identity implementation needs a big refactor
+
+// TODO(unassigned): keep an eye open for alignment issues and for hacks like:
+// copy public and signing keys together
+//memcpy(public_key, keys.public_key, sizeof(public_key) + sizeof(signing_key));
+
 Identity& Identity::operator=(const Keys& keys) {
-  // copy public and signing keys together
-  memcpy(public_key, keys.public_key, sizeof(public_key) + sizeof(signing_key));
+  memcpy(public_key, keys.public_key, sizeof(public_key));
```

Comment:
We should still support DSA-SHA1 for now (to be able to communicate with old routers), I think. But it certainly shouldn't be the default signature type for anything in the kovri router.